File: data/code_pairs/cpp/280.cpp
#include <iostream>

namespace foo {
    struct t {
        int i;
        t(int val) : i(val) {}  // Constructor to allow initialization
    };

    int operator*(const t& lhs, const t& rhs) {
        return lhs.i * rhs.i;
    }
}

namespace bar {
    struct t2 {
        int i;
        t2(int val) : i(val) {}  // Constructor to allow initialization
    };

    bool operator>(const t2& lhs, const t2& rhs) {
        return lhs.i > rhs.i;
    }
}

int main() {
    foo::t i(1), j(2);
    bar::t2 k(1), l(2);

    std::cout << "i*j = " << i * j << std::endl;
    std::cout << "k > l = " << (k > l ? "true" : "false") << std::endl;

    return 0;
}

File: data/code_pairs/fortran/280.f90
module foo
  implicit none
  type :: t
     integer :: i
  end type t
  interface operator (*)
     module procedure mult
  end interface
contains
  function mult(i, j)
    type(t), intent(in) :: i, j
    integer             :: mult
    mult = i%i * j%i
  end function mult
end module foo

module bar
  implicit none
  type :: t2
     integer :: i
  end type t2
  interface operator (>)
     module procedure gt
  end interface
contains
  function gt(i, j)
    type(t2), intent(in) :: i, j
    logical               :: gt
    gt = i%i > j%i
  end function gt
end module bar

program testDriver
  use foo
  use bar
  implicit none
  type(t) :: i = t(1), j = t(2)
  type(t2) :: k = t2(1), l = t2(2)

  print *, 'i*j = ', i*j
  print *, 'k > l = ', k > l
end program testDriver